File: app/src/main/java/com/druger/aboutwork/di/modules/NetworkModule.java
package com.druger.aboutwork.di.modules;

import com.druger.aboutwork.rest.RestApi;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by druger on 22.07.2017.
 */

@Module
public class NetworkModule {

    private RestApi restApi;

    public NetworkModule() {
        restApi = new RestApi();
    }

    @Singleton
    @Provides
    RestApi provideApiClient() {
        return restApi;
    }
}

File: app/src/main/java/com/druger/aboutwork/di/modules/NetworkModule.kt
package com.druger.aboutwork.di.modules

import com.druger.aboutwork.rest.RestApi
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by druger on 22.07.2017.
 */

@Module
class NetworkModule {

    private val restApi: RestApi = RestApi()

    @Singleton
    @Provides
    internal fun provideApiClient(): RestApi {
        return restApi
    }
}
